Examples/ObjectiveC/UserTrackingModesExample.m

```diff
@@ -3,23 +3,182 @@
+}
+
+- (void)layoutArrow {
+    if (_arrow == nil) {
+        
+        CAShapeLayer *arrow = [[CAShapeLayer alloc] init];
+        
+        arrow.path = [self arrowPath];
+        arrow.lineWidth = 2;
+        arrow.lineJoin = kCALineJoinRound;
+        arrow.bounds = CGRectMake(0, 0, UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+        arrow.position = CGPointMake(UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+        arrow.shouldRasterize = YES;
+        arrow.rasterizationScale = [[UIScreen mainScreen] scale];
+        arrow.drawsAsynchronously = YES;
+
+        _arrow = arrow;
+        
+        [self updateArrow:MGLUserTrackingModeNone];
+        [self.layer addSublayer:_arrow];
+    }
+}
+
+- (CGPathRef) arrowPath {
+    
+    CGFloat max = UserLocationButtonSize / 2;
+    
+    UIBezierPath *bezierPath = [[UIBezierPath alloc] init];
+    
+    [bezierPath moveToPoint:CGPointMake(max * 0.5, 0)];
+    
+    [bezierPath addLineToPoint:CGPointMake(max * 0.1, max)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, max * 0.65)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.9, max)];
+    [bezierPath addLineToPoint:CGPointMake(max * 0.5, 0)];
+    [bezierPath closePath];
+    
+    return bezierPath.CGPath;
+}
+
+-(void)updateArrow:(MGLUserTrackingMode)mode {
+    CGColorRef arrowStrokeColor;
+    CGPoint arrowPosition;
+    CGColorRef arrowFillColor;
+    CGFloat arrowRotation;
+    
+    switch (mode) {
+        case MGLUserTrackingModeNone:
+            arrowStrokeColor = [[UIColor whiteColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+            arrowFillColor = [[UIColor clearColor] CGColor];
+            arrowRotation = 0;
+            break;
+        case MGLUserTrackingModeFollow:
+            arrowStrokeColor = self.tintColor.CGColor;
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2 + 2, UserLocationButtonSize / 2 - 2);
+            arrowFillColor = [[UIColor clearColor] CGColor];
+            arrowRotation = 0.66;
+            break;
+        case MGLUserTrackingModeFollowWithHeading:
+            arrowStrokeColor = [[UIColor clearColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2 + 2, UserLocationButtonSize / 2 - 2);
+            arrowFillColor = self.tintColor.CGColor;
+            arrowRotation = 0.66;
+            break;
+        case MGLUserTrackingModeFollowWithCourse:
+            arrowStrokeColor = [[UIColor clearColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+            arrowFillColor = self.tintColor.CGColor;
+            arrowRotation = 0;
+            break;
+    }
+    
+    [_arrow setFillColor:arrowFillColor];
+    [_arrow setStrokeColor:arrowStrokeColor];
+    [_arrow setAffineTransform:CGAffineTransformMakeRotation(arrowRotation)];
+    
+    [self layoutIfNeeded];
+}
+
+@end
+
```

Comment:
Add a comment/marker here to better delineate the two classes.

Reply:
The special [`#pragma mark -`](http://nshipster.com/pragma/) would be a good choice, as Xcode understands this as an explicit section separator.